### PR TITLE
added missing fields

### PR DESCRIPTION
**Pull request by trapacska**
Deploy to Bitrise.io step uploaded an xml in which the corresponding fields were missing. With this fix those fields are passed through also in a way that all the error fields are merged into a single failure field. See the ticket for more technical details.

### CODE REVIEW

File: test/converters/junitxml/junitxml.go [hunk 1]
```diff
@@ -25,6 +25,44 @@ func (h *Converter) Detect(files []string) bool {
 	return len(h.files) > 0
 }
 
+// merges Suites->Cases->Error and Suites->Cases->SystemErr field values into Suites->Cases->Failure field
+// with 2 newlines and error category prefix
+// the two newlines applied only if there is a failur emessage already
+func regroupErrors(suites []junit.TestSuite) []junit.TestSuite {
```
Comment (richard-bogdan-bitrise) [inline test/converters/junitxml/junitxml.go]: Please add a unit test for this.

File: test/converters/junitxml/junitxml.go [hunk 2]
```diff
@@ -25,6 +25,44 @@ func (h *Converter) Detect(files []string) bool {
 	return len(h.files) > 0
 }
 
+// merges Suites->Cases->Error and Suites->Cases->SystemErr field values into Suites->Cases->Failure field
+// with 2 newlines and error category prefix
+// the two newlines applied only if there is a failure message already
+func regroupErrors(suites []junit.TestSuite) []junit.TestSuite {
+	for testSuiteIndex := range suites {
+		for testCaseIndex := range suites[testSuiteIndex].TestCases {
+			tc := suites[testSuiteIndex].TestCases[testCaseIndex]
```
Comment (godrei) [inline test/converters/junitxml/junitxml.go]: NOTE: using these for loops you can spare this line:

```
	for testSuiteIndex, testSuit := range suites {
		for testCaseIndex, tc := range testSuit.TestCases {

		}
	}
```

File: test/converters/junitxml/junitxml.go [hunk 3]
```diff
@@ -34,15 +72,15 @@ func parseTestSuites(filePath string) ([]junit.TestSuite, error) {
 	var testSuites junit.XML
 	testSuitesError := xml.Unmarshal(data, &testSuites)
 	if testSuitesError == nil {
-		return testSuites.TestSuites, nil
+		return regroupErrors(testSuites.TestSuites), nil
 	}
 
 	var testSuite junit.TestSuite
 	if err := xml.Unmarshal(data, &testSuite); err != nil {
 		return nil, errors.Wrap(errors.Wrap(err, string(data)), testSuitesError.Error())
```
Comment (godrei) [inline test/converters/junitxml/junitxml.go]: why do you include the `testSuitesError` in the `parseTestSuites ` function's error message?
i feel `parseTestSuites` error should only describe why test suit parsing failed.

File: test/converters/junitxml/junitxml_test.go [hunk 4]
```diff
@@ -23,3 +26,166 @@ func Test_parseTestSuites(t *testing.T) {
 		})
 	}
 }
+
+func Test_regroupErrors(t *testing.T) {
```
Comment (godrei) [inline test/converters/junitxml/junitxml_test.go]: would not it be easier to understand by having xml content test cases?

File: test/junit/xml.go [hunk 5]
```diff
@@ -26,4 +26,13 @@ type TestCase struct {
 	ClassName string   `xml:"classname,attr"`
 	Time      float64  `xml:"time,attr"`
 	Failure   string   `xml:"failure,omitempty"`
+	Error     *Error   `xml:"error,omitempty"`
```
Comment (godrei) [inline test/junit/xml.go]: NOTE: would not it better to have separate models for our junit representation and for the client representations?

File: test/converters/junitxml/junitxml.go (hunk 3 above)
Comment (trapacska) [reply test/converters/junitxml/junitxml.go]: because of the fallback, the first error message might can tell something

**Pull request merged**